### PR TITLE
Update copyright notices to 2023

### PR DESCRIPTION
**Pull request by cqc-alec**

### CODE REVIEW

File: .github/workflows/compare-coverage [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 
-# Copyright 2022 Cambridge Quantum Computing
+# Copyright 2019-2023 Cambridge Quantum Computing
```
Comment (cqc-melf) [inline .github/workflows/compare-coverage:3]: Is this intended or should this be 2022-2023?

Comment (cqc-alec) [reply .github/workflows/compare-coverage:3]: This is intended; there were two files which had this notice but I decided to change them all to the same -- makes search-and-replace easier in the future. We don't generally track the start date of each file individually.

Comment (cqc-alec) [reply .github/workflows/compare-coverage:3]: Apart from those two files in this directory the only other change not covered by the search-and-replace was to `pytket/docs/conf.py`.

Comment (cqc-melf) [reply .github/workflows/compare-coverage:3]: Sounds good!

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**